src/qr.jl

```diff
@@ -538,8 +538,9 @@ end
 function rank(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(float(one(eltype(A.Q))))) * iszero(atol))
     m = min(size(A)...)
     m == 0 && return 0
-    tol = max(atol, rtol*abs(A.R[1,1]))
-    return something(findfirst(i -> abs(A.R[i,i]) <= tol, 1:m), m+1) - 1
+    factors = getfield(A, :factors)
```

Comment:
Why not just `A.factors` instead of the `getfield` call?

Reply:
I went with `getfield` because its more direct. Calling `A.factors` will first call `getproperty` go through an `if .. else` ladder and then call `getfield(A, :factors)` anyways. I just thought it would be cleaner also because `factors` is not a property that shows up `propertynames`. They both perform similarly fwiw

Reply:
I think it's fine to call `A.factors`, as constant propagation will inline the correct branch and replace it with the corresponding `getfield` call. It's just a personal preference though, at the end of it, as calling `getfield` directly feels too low-level and implementation-specific to me.

Reply:
Yes, normally the only reason to call `getfield` is if `getproperty` doesn't give you access to the field.